projects/igniteui-angular/src/lib/directives/focus-trap/focus-trap.directive.ts

```diff
@@ -0,0 +1,101 @@
+import { AfterViewInit, ContentChild, Directive, ElementRef, HostBinding, Input, NgModule, ViewChild } from "@angular/core";
+import { IgxToggleDirective } from "igniteui-angular";
+import { PlatformUtil } from "../../core/utils";
+
+@Directive({
+    selector: '[igxFocusTrap]'
+})
+export class IgxFocusTrapDirective implements AfterViewInit {
+    /**
+     * @hidden
+     */
+    public get element(): HTMLElement | null {
+        return this.elementRef.nativeElement;
+    }
+
+    /** @hidden */
+    public ngAfterViewInit(): void {
+        const modal = this.element.querySelector('.igx-toggle--hidden') as HTMLElement;
+        modal.addEventListener('keydown', event => {
+            if (event.key === this.platformUtil.KEYMAP.TAB) {
+                this.handleTab(event);
+            }
+        });
+    }
+
+    /**
+     * @hidden
+     */
+    constructor(
+        private elementRef: ElementRef,
+        protected platformUtil: PlatformUtil) {
+    }
+
+    private get target(): any {
+        return this.overlay.querySelector('.igx-toggle');
+    }
+
+    private get overlay(): Element {
+        return document.querySelector('.igx-overlay');
+    }
+
+    private handleTab(event) {
+        const elements = this.getFocusableElements(this.target);
```

Comment:
we can greatly simplify this logic. Once we have all the `focusableElements` we can determine, depending on tab direction, which is the next focusable element index. Having the next element index we should call focus on this element.